Source/Scene/GroundPolylinePrimitive.js

```diff
@@ -530,25 +577,25 @@ define([
             boundingSpheres = primitive._boundingSphereMorph;
         }
 
-        var commandList = frameState.commandList;
+        var morphing = frameState.mode === SceneMode.MORPHING;
+        var classificationType = groundPolylinePrimitive.classificationType;
+        var queueTerrainCommands = (classificationType !== ClassificationType.CESIUM_3D_TILE);
+        var queue3DTilesCommands = (classificationType !== ClassificationType.TERRAIN) && !morphing;
+
+        var command;
         var passes = frameState.passes;
         if (passes.render || (passes.pick && primitive.allowPicking)) {
             var colorLength = colorCommands.length;
-
             for (var j = 0; j < colorLength; ++j) {
-                var colorCommand = colorCommands[j];
-                // Use derived appearance command for morph and 2D
-                if (frameState.mode === SceneMode.MORPHING && colorCommand.shaderProgram !== groundPolylinePrimitive._spMorph) {
-                    colorCommand = colorCommand.derivedCommands.colorMorph;
-                } else if (frameState.mode !== SceneMode.SCENE3D && colorCommand.shaderProgram !== groundPolylinePrimitive._sp2D) {
-                    colorCommand = colorCommand.derivedCommands.color2D;
+                var boundingVolume = boundingSpheres[j];
+                if (queueTerrainCommands) {
+                    command = colorCommands[j];
+                    updateAndQueueCommand(groundPolylinePrimitive, command, frameState, modelMatrix, cull, boundingVolume, debugShowBoundingVolume);
+                }
+                if (queue3DTilesCommands) {
+                    command = colorCommands[j].derivedCommands.tileset;
+                    updateAndQueueCommand(groundPolylinePrimitive, command, frameState, modelMatrix, cull, boundingVolume, debugShowBoundingVolume);
```

Comment:
Wouldn't this throw in 2D? The tileset command doesn't have any derived commands.

Reply:
The 2D derived command is created here: https://github.com/AnalyticalGraphicsInc/cesium/pull/7437/files#diff-d3380c9fefd2c0cccbc42b72efa86ce8R536